crates/net/network/src/transactions/mod.rs

```diff
@@ -915,6 +917,9 @@ where
         let mut num_already_seen = 0;
 
         if let Some(peer) = self.peers.get_mut(&peer_id) {
+            // pre-size to avoid reallocations, assuming ~50% of the transactions are new
+            let mut to_add = Vec::with_capacity(max(1, transactions.len() / 2));
```

Comment:
it's probably a good idea to keep this in the local scope like has been done. otherwise, if `to_add` persists between calls to make sure the batch is a substantial size before submitting to the pool, the risk is the pool waits too long for certain transactions.

Reply:
All queued transaction events are drained before the `pool_imports`, so in theory accumulating all the new txs in a single vec, and doing a single batch-add at the end should be equivalent to how it's done currently? Because all this work is driven by the same future (`TransactionManager`), the first transaction event in the queue will be processed at the same time as all other tx events.

We can get rid of the `pool_imports`altogether and just pass a mutable vec to each `import_transactions` call in this case. LMK if you'd like me to proceed with this change @emhane

Reply:
Actually, currently when processing imports the batch add future could yield, so this will probably change the semantics.